extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -28,42 +25,32 @@ import { logger } from './logging';
  */
 export async function promptImportInternetDatabase(
   databasesManager: DatabaseManager,
-  storagePath: string
+  storagePath: string,
+  progress: ProgressCallback,
+  _: CancellationToken,
 ): Promise<DatabaseItem | undefined> {
-  let item: DatabaseItem | undefined = undefined;
-
-  try {
-    const databaseUrl = await window.showInputBox({
-      prompt: 'Enter URL of zipfile of database to download',
-    });
-    if (databaseUrl) {
-      validateHttpsUrl(databaseUrl);
-
-      const progressOptions: ProgressOptions = {
-        location: ProgressLocation.Notification,
-        title: 'Adding database from URL',
-        cancellable: false,
-      };
-      await withProgress(
-        progressOptions,
-        async (progress) =>
-          (item = await databaseArchiveFetcher(
-            databaseUrl,
-            databasesManager,
-            storagePath,
-            progress
-          ))
-      );
-      if (item) {
-        commands.executeCommand('codeQLDatabases.focus');
-        showAndLogInformationMessage('Database downloaded and imported successfully.');
-      }
-    }
-  } catch (e) {
-    showAndLogErrorMessage(e.message);
+  const databaseUrl = await window.showInputBox({
+    prompt: 'Enter URL of zipfile of database to download',
+  });
+  if (!databaseUrl) {
+    return;
   }
 
+  validateHttpsUrl(databaseUrl);
+
+  const item = await databaseArchiveFetcher(
+    databaseUrl,
+    databasesManager,
+    storagePath,
+    progress
```

Comment:
Can we use the cancel token here to make the operation cancellable?

Reply:
Not so easy. The download and unzip functions are from third parties and do not easily accept cancellations.